### DISCUSSION TITLE
Side by Side Experimental Feature Feedback

### DISCUSSION THREAD

**Discussion by JohnMcPMS** [microsoft/winget-cli#4281] [General]
Please provide feedback on the side by side experimental feature here.  Anything from issues with the functionality, ideas on how to better display the side by side items in results, or improvements to the command arguments.

PR for reference: #4282

**Comment by Schtenk** [comment]
Shouid this feature also handle the scenario of when both the x86/x64(with the same id) versions of a package are installed?

`Microsoft.DotNet.DesktopRuntime.6` (x86/x64) is good example for when this can happen, when different programs sometimes require one or the other(even though only one of them should be required right?) 

The current experimental version does not seem to work for/fix this scenario.

**Comment by eabase** [comment]
`winget list --upgrade-available` not working after upgrade to latest *preview* (to `v1.8.1133-preview` from `v1.7.11132`).

Before
```powershell
# winget list --upgrade-available
Name                                             Id                                Version  Available Source
------------------------------------------------------------------------------------------------------------
Microsoft Windows Desktop Runtime - 6.0.20 (x86) Microsoft.DotNet.DesktopRuntime.6 6.0.20   6.0.29    winget
Python Launcher                                  Python.Launcher                   < 3.12.0 3.12.0    winget
2 upgrades available.
```

After
```powershell
# winget list --upgrade-available
No installed package found matching input criteria.
```

**Comment by aaronliu0130** [comment]
See also #4357

**Comment by mdanish-kh** [comment]
A new parameter `--versions` for `winget list` command to display all side-by-side installed versions of a package.

Expected output:

```
~ winget list Microsoft.PowerShell --versions
Found PowerShell [Microsoft.PowerShell]
Version
--------
7.4.3.0
7.4.2.0
```

One use case, among others, is that this will help users select the version for the `winget uninstall --version <Version>` and offer a familiar UX as `winget (search|show) --versions`
